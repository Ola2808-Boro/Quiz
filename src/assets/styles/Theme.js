export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
