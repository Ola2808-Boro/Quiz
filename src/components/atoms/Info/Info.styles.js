import { device } from "helpers/breakpoints";
import styled from "styled-components";

export const StyledFooter = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 1;

  p {
    font-size: ${({ theme }) => theme.fontSize.m};

    @media ${device.mobileM} {
      font-size: ${({ theme }) => theme.fontSize.ml};
    }
  }
  a {
    text-decoration: none;
    letter-spacing: 1px;
  }
`;
