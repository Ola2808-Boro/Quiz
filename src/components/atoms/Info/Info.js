import React from "react";
import { StyledFooter } from "./Info.styles";

const Info = () => {
  return (
    <StyledFooter>
      <p>
        Developed by{" "}
        <a
          href="https://github.com/Ola2808-Boro"
          target="_blank"
          rel="noreferrer"
        >
          Alekandra Borowska
        </a>
      </p>
    </StyledFooter>
  );
};

export default Info;
