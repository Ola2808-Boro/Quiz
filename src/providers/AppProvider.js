import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;