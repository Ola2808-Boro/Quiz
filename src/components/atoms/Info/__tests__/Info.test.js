import "jest-styled-components";
import renderer from "react-test-renderer";
import AppProvider from "providers/AppProvider";
import React from "react";
import { StyledFooter } from "components/atoms/Info/Info.styles";
import Info from "components/atoms/Info/Info";
import { render, screen } from "@testing-library/react";
describe("Test info component", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppProvider>
          <StyledFooter />
        </AppProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("test rendering", () => {
    render(
      <AppProvider>
        <Info />
      </AppProvider>
    );
    const infoElment = screen.getByText(/developed by/i);
    expect(infoElment).toBeInTheDocument();
  });
});
