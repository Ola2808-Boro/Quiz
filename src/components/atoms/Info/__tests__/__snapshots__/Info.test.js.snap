// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test info component default style 1`] = `
<footer
  className="iLOoBX"
/>
`;
